src/webui/www/public/scripts/login.js

```diff
@@ -30,32 +30,32 @@
 
 const submitLoginForm = (event) => {
     event.preventDefault();
-    const errorMsgElement = document.getElementById("error_msg");
 
-    const xhr = new XMLHttpRequest();
-    xhr.open("POST", "api/v2/auth/login", true);
-    xhr.setRequestHeader("Content-type", "application/x-www-form-urlencoded; charset=UTF-8");
-    xhr.addEventListener("readystatechange", () => {
-        if (xhr.readyState === 4) { // DONE state
-            if ((xhr.status === 200) && (xhr.responseText === "Ok."))
-                location.replace(location);
-            else
-                errorMsgElement.textContent = "QBT_TR(Invalid Username or Password.)QBT_TR[CONTEXT=Login]";
-        }
-    });
-    xhr.addEventListener("error", () => {
-        errorMsgElement.textContent = (xhr.responseText !== "")
-            ? xhr.responseText
-            : "QBT_TR(Unable to log in, server is probably unreachable.)QBT_TR[CONTEXT=Login]";
-    });
+    const errorMsgElement = document.getElementById("error_msg");
+    errorMsgElement.textContent = ""; // clear previous error
 
     const usernameElement = document.getElementById("username");
     const passwordElement = document.getElementById("password");
-    const queryString = "username=" + encodeURIComponent(usernameElement.value) + "&password=" + encodeURIComponent(passwordElement.value);
-    xhr.send(queryString);
+    const queryString = `username=${encodeURIComponent(usernameElement.value)}&password=${encodeURIComponent(passwordElement.value)}`;
```

Comment:
`URLSearchParams` will handle this complexity for you.

```
const query = new URLSearchParams();
query.append("username", usernameElement.value);
query.append("password", passwordElement.value);
// body: query.toString()
```

Reply:
Thanks, done.